tests/shared.py

```diff
@@ -24,16 +24,28 @@ def gradient(self, symbol, discretised_symbol, boundary_conditions):
         n = 0
         for domain in symbol.domain:
             n += self.mesh[domain][0].npts
-        gradient_matrix = pybamm.Matrix(np.eye(n))
+        gradient_matrix = pybamm.Matrix(eye(n))
         return gradient_matrix @ discretised_symbol
 
     def divergence(self, symbol, discretised_symbol, boundary_conditions):
         n = 0
         for domain in symbol.domain:
             n += self.mesh[domain][0].npts
-        divergence_matrix = pybamm.Matrix(np.eye(n))
+        divergence_matrix = pybamm.Matrix(eye(n))
         return divergence_matrix @ discretised_symbol
 
+    def compute_diffusivity(
+        self, symbol, extrapolate_left=None, extrapolate_right=None
+    ):
+        return symbol
```

Comment:
can leave this out to take default behaviour from the base `SpatialMethod` class